Branch-SDK/Branch-SDK/Networking/Requests/BranchOpenRequest.m

```diff
@@ -232,16 +232,16 @@ - (void)processResponse:(BNCServerResponse *)response error:(NSError *)error {
     }
 
     NSString *referringURL = nil;
-    if (preferenceHelper.universalLinkUrl) {
+    if (preferenceHelper.universalLinkUrl.length) {
         referringURL = preferenceHelper.universalLinkUrl;
     }
-    else if (preferenceHelper.externalIntentURI) {
+    else if (preferenceHelper.externalIntentURI.length) {
         referringURL = preferenceHelper.externalIntentURI;
     }
     else {
         NSDictionary *sessionDataDict = [BNCEncodingUtils decodeJsonStringToDictionary:sessionData];
         NSString *link = sessionDataDict[BRANCH_RESPONSE_KEY_BRANCH_REFERRING_LINK];
-        if (link) referringURL = link;
+        if (link.length) referringURL = link;
```

Comment:
what if `link` is `nil`? Would that cause runtime errors?

Reply:
No. In obj-c, sending a message to a nil object has no effect.